desktop/core/src/desktop/lib/botserver/views.py

```diff
@@ -161,12 +167,16 @@ def _make_unfurl_payload(url, id_type, doc, doc_type):
   request = MockRequest(user=user)
 
   if id_type == 'editor':
+    # Max row and column allowed in result preview table
+    max_cols = 4
+    max_rows = 20
+
     try:
       status = _check_status(request, operation_id=doc_data['uuid'])
       if status['query_status']['status'] == 'available':
-        fetch_result = query_result(request, json.loads(doc.data))
-        unfurl_result = _make_result_table(fetch_result) if isinstance(fetch_result, dict) else fetch_result
-    except:
+        fetch_result = query_result(request, json.loads(doc.data), max_rows)
+        unfurl_result = _make_result_table(fetch_result, max_cols) if isinstance(fetch_result, dict) else fetch_result
```

Comment:
Don't do it there, just keep the logic in one place?
e.g.

1. we get the data
2. we truncate the data
3. we just send it to prettify

Reply:
Agreed! Motive of the function should be clean!
Fixed